test/unit/app/repos/dogs.test.js

```diff
@@ -724,4 +725,81 @@ describe('Dog repo', () => {
       })
     })
   })
+
+  describe('constructStatusList', () => {
+    test('should construct correct status id list if more than one status', async () => {
+      const statusList = 'Pre-exempt,Exempt,In breach'
+
+      const res = await constructStatusList(statusList)
+
+      expect(res).toEqual([2, 3, 5])
+    })
+
+    test('should construct correct status id list if only one status', async () => {
+      const statusList = 'Failed'
+
+      const res = await constructStatusList(statusList)
+
+      expect(res).toEqual([4])
+    })
+
+    test('should construct correct status id list if only one status', async () => {
+      const res = await constructStatusList(null)
+
+      expect(res).toEqual([])
+    })
+  })
+
+  describe('constructDbSort', () => {
+    test('should construct default sort construct when no params supplied', async () => {
+      sequelize.col.mockReturnValue((column) => column)
+      sequelize.literal = jest.fn()
+
+      const res = constructDbSort(null, [1, 2])
+
+      expect(res.length).toBe(2)
+    })
+
+    test('should construct correct sort construct', async () => {
+      sequelize.col.mockReturnValue((column) => column)
+
+      let res = constructDbSort({ sortOrder: 'DESC', sortKey: 'cdoIssued' })
+      expect(res).toEqual([[sequelize.col('dog.index_number'), 'DESC']])
+
+      res = constructDbSort({ sortOrder: 'DESC', sortKey: 'indexNumber' })
+      expect(res).toEqual([[sequelize.col('dog.index_number'), 'DESC']])
+
+      res = constructDbSort({ sortOrder: 'DESC', sortKey: 'dateOfBirth' })
+      expect(res).toEqual([[sequelize.col('dog.index_number'), 'DESC']])
+
+      res = constructDbSort({ sortOrder: 'ASC', sortKey: 'selected' })
+      expect(res).toEqual([])
+    })
+  })
+
+  describe('getOldDogs', () => {
+    test('should construct correct correct where clause', async () => {
+      sequelize.models.registration.findAll.mockResolvedValue()
+      const statusList = 'Pre-exempt,Exempt'
+
+      await getOldDogs(statusList)
+
+      // console.log('here1', sequelize.models.registration.findAll.mock.calls[0][0].where)
```

Comment:
did you want to keep this?

Reply:
I've redone that test - slight crossover in messages